shell/platform/android/test/io/flutter/embedding/android/AndroidTouchProcessorTest.java

```diff
@@ -218,4 +218,58 @@ public void unexpectedMaskedAction() {
     touchProcessor.onTouchEvent(mocker.mockEvent(MotionEvent.ACTION_BUTTON_PRESS, 0.0f, 0.0f, 0));
     verify(mockRenderer, never()).dispatchPointerDataPacket(ByteBuffer.allocate(0), 0);
   }
+
+  @Test
+  public void unexpectedPointerChange() {
+    // Regression test for https://github.com/flutter/flutter/issues/129765
+
+    MotionEventMocker mocker =
+        new MotionEventMocker(0, InputDevice.SOURCE_MOUSE, MotionEvent.TOOL_TYPE_MOUSE);
+
+    touchProcessor.onTouchEvent(mocker.mockEvent(MotionEvent.ACTION_DOWN, 0.0f, 0.0f, 0));
+    InOrder inOrder = inOrder(mockRenderer);
+    inOrder
+        .verify(mockRenderer)
+        .dispatchPointerDataPacket(packetCaptor.capture(), packetSizeCaptor.capture());
+    ByteBuffer packet = packetCaptor.getValue();
+    assertEquals(AndroidTouchProcessor.PointerChange.PAN_ZOOM_START, readPointerChange(packet));
+    assertEquals(AndroidTouchProcessor.PointerDeviceKind.TRACKPAD, readPointerDeviceKind(packet));
+    assertEquals(AndroidTouchProcessor.PointerSignalKind.NONE, readPointerSignalKind(packet));
+    assertEquals(0.0, readPointerPhysicalX(packet));
+    assertEquals(0.0, readPointerPhysicalY(packet));
+
+    touchProcessor.onTouchEvent(mocker.mockEvent(MotionEvent.ACTION_MOVE, 10.0f, 5.0f, 0));
+    inOrder
+        .verify(mockRenderer)
+        .dispatchPointerDataPacket(packetCaptor.capture(), packetSizeCaptor.capture());
+    packet = packetCaptor.getValue();
+    assertEquals(AndroidTouchProcessor.PointerChange.PAN_ZOOM_UPDATE, readPointerChange(packet));
+    assertEquals(AndroidTouchProcessor.PointerDeviceKind.TRACKPAD, readPointerDeviceKind(packet));
+    assertEquals(AndroidTouchProcessor.PointerSignalKind.NONE, readPointerSignalKind(packet));
+    assertEquals(0.0, readPointerPhysicalX(packet));
+    assertEquals(0.0, readPointerPhysicalY(packet));
+    assertEquals(10.0, readPointerPanX(packet));
+    assertEquals(5.0, readPointerPanY(packet));
+
+    touchProcessor.onGenericMotionEvent(mocker.mockEvent(MotionEvent.ACTION_SCROLL, 0.0f, 0.0f, 0));
+    inOrder
+        .verify(mockRenderer)
+        .dispatchPointerDataPacket(packetCaptor.capture(), packetSizeCaptor.capture());
+    packet = packetCaptor.getValue();
+    assertEquals(0.0, readPointerPhysicalX(packet));
```

Comment:
Can you test, what are the deviceKind, signalKind, and pointerChange here?

Reply:
Done